cmd/osv-scanner/main.go

```diff
@@ -35,211 +34,148 @@ func run(args []string, stdout, stderr io.Writer) int {
-			if context.IsSet("experimental-licenses") {
-				if len(allowlist) == 0 ||
-					(len(allowlist) == 1 && allowlist[0] == "") {
-					return fmt.Errorf("--experimental-licenses requires at least one value")
-				}
-				if unrecognized := spdx.Unrecognized(allowlist); len(unrecognized) > 0 {
-					return fmt.Errorf("--experimental-licenses requires comma-separated spdx licenses. The following license(s) are not recognized as spdx: %s", strings.Join(unrecognized, ","))
-				}
-			}
-
-			verbosityLevel, err := reporter.ParseVerbosityLevel(context.String("verbosity"))
-			if err != nil {
-				return err
-			}
-
-			if r, err = reporter.New(format, stdout, stderr, verbosityLevel, termWidth); err != nil {
-				return err
-			}
-
-			var callAnalysisStates map[string]bool
-			if context.IsSet("experimental-call-analysis") {
-				callAnalysisStates = createCallAnalysisStates([]string{"all"}, context.StringSlice("no-call-analysis"))
-				r.Infof("Warning: the experimental-call-analysis flag has been replaced. Please use the call-analysis and no-call-analysis flags instead.\n")
-			} else {
-				callAnalysisStates = createCallAnalysisStates(context.StringSlice("call-analysis"), context.StringSlice("no-call-analysis"))
-			}
-
-			vulnResult, err := osvscanner.DoScan(osvscanner.ScannerActions{
-				LockfilePaths:        context.StringSlice("lockfile"),
-				SBOMPaths:            context.StringSlice("sbom"),
-				DockerContainerNames: context.StringSlice("docker"),
-				Recursive:            context.Bool("recursive"),
-				SkipGit:              context.Bool("skip-git"),
-				NoIgnore:             context.Bool("no-ignore"),
-				ConfigOverridePath:   context.String("config"),
-				DirectoryPaths:       context.Args().Slice(),
-				CallAnalysisStates:   callAnalysisStates,
-				ExperimentalScannerActions: osvscanner.ExperimentalScannerActions{
-					LocalDBPath:    context.String("experimental-local-db-path"),
-					CompareLocally: context.Bool("experimental-local-db"),
-					CompareOffline: context.Bool("experimental-offline"),
-					// License summary mode causes all
-					// packages to appear in the json as
-					// every package has a license - even
-					// if it's just the UNKNOWN license.
-					ShowAllPackages: context.Bool("experimental-all-packages") ||
-						context.Bool("experimental-licenses-summary"),
-					ScanLicensesSummary:   context.Bool("experimental-licenses-summary"),
-					ScanLicensesAllowlist: context.StringSlice("experimental-licenses"),
-				},
-			}, r)
-
-			if err != nil && !errors.Is(err, osvscanner.VulnerabilitiesFoundErr) {
-				return err
-			}
-
-			if errPrint := r.PrintResult(&vulnResult); errPrint != nil {
-				return fmt.Errorf("failed to write output: %w", errPrint)
-			}
+	}
 
-			// This may be nil.
-			return err
-		},
+	if len(args) >= 2 {
+		if args[1] != "scan" && args[1] != "update" && args[1] != "fix" && args[1] != "--help" && args[1] != "--version" {
```

Comment:
Add these against to an array variable, and populate that variable with a function that loops through the top level commands in the cli struct to make sure these 2 are always synced.